src/components/Quiz/utils.ts

```diff
@@ -0,0 +1,134 @@
+import { getLocaleForNumberFormat } from "../../utils/translations"
+import { Lang } from "../../utils/languages"
+
+import { CompletedQuizzes, QuizShareStats } from "../../types"
+
+import {
+  TOTAL_QUIZ_AVERAGE_SCORE,
+  TOTAL_QUIZ_QUESTIONS_ANSWERED,
+  TOTAL_QUIZ_RETRY_RATE,
+  USER_STATS_KEY,
+} from "../../constants"
+
+import allQuizzesData, {
+  ethereumBasicsQuizzes,
+  usingEthereumQuizzes,
+} from "../../data/quizzes"
+
+export const getTotalQuizzesPoints = () =>
+  Object.values(allQuizzesData)
+    .map((quiz) => quiz.questions.length)
+    .reduce((accumulator, currentValue) => {
+      return accumulator + currentValue
+    }, 0)
+
+export const getNumberOfCompletedQuizzes = (quizzes: CompletedQuizzes) =>
+  Object.values(quizzes)
+    .map((v) => v[0])
+    .filter((v) => v).length
+
+export const getNextQuiz = (currentQuiz?: string) => {
+  const allQuizzes = [...ethereumBasicsQuizzes, ...usingEthereumQuizzes]
+  const nextQuiz = allQuizzes.find((quiz) => quiz.id === currentQuiz)
+
+  return nextQuiz ? nextQuiz.next : undefined
+}
```

Comment:
`ethereumBasicsQuizzes` and `usingEthereumQuizzes` are both ordered arrays... is there a reason we don't just update this based on the next one in the array to avoid having to maintain the `next` property?

Reply:
@wackerow good observation, I think having a `next` pointer simplifies the implementation (in my head this was like a list of nodes), otherwise you need some extra logic work with the indexes